src/phoenix/db/insertion/dataset.py

```diff
@@ -0,0 +1,244 @@
+            name = await _get_unique_name(session)
+        else:
+            raise ValueError(f"Dataset name must not be empty for action={action.value}")
+    keys = DatasetKeys(frozenset(input_keys), frozenset(output_keys), frozenset(metadata_keys))
+    created_at = datetime.now(timezone.utc)
+    dataset_id: Optional[DatasetId] = None
+    if action is DatasetTableAction.APPEND and name:
+        dataset_id = await session.scalar(
+            select(models.Dataset.id).where(models.Dataset.name == name)
+        )
+    if action is DatasetTableAction.CREATE or dataset_id is None:
+        try:
+            dataset_id = await insert_dataset(
+                session=session,
+                name=name,
+                description=description,
+                metadata=metadata,
+                created_at=created_at,
+            )
+        except Exception:
+            logger.exception(
+                f"Fail to insert dataset: {input_keys=}, {output_keys=}, {metadata_keys=}"
+            )
+            raise
+    assert dataset_id is not None
+    try:
+        dataset_version_id = await insert_dataset_version(
+            session=session,
+            dataset_id=dataset_id,
+            created_at=created_at,
+        )
+    except Exception:
+        logger.exception(f"Fail to insert dataset version for {dataset_id=}")
+        raise
+    assert dataset_version_id is not None
+    for row in table:
+        try:
+            dataset_example_id = await insert_dataset_example(
+                session=session,
+                dataset_id=dataset_id,
+                created_at=created_at,
+            )
+        except Exception:
+            logger.exception(f"Fail to insert dataset example for {dataset_id=}")
+            raise
+        assert dataset_example_id is not None
+        try:
+            dataset_example_revision = await insert_dataset_example_revision(
+                session=session,
+                dataset_version_id=dataset_version_id,
+                dataset_example_id=dataset_example_id,
+                input={key: row.get(key) for key in keys.input},
+                output={key: row.get(key) for key in keys.output},
+                metadata={key: row.get(key) for key in keys.metadata},
+                created_at=created_at,
+            )
+        except Exception:
+            logger.exception(
+                f"Fail to insert dataset example revision for {dataset_version_id=}, "
+                f"{dataset_example_id=}"
+            )
+            raise
+        assert dataset_example_revision is not None
+    return DatasetCreationEvent(dataset_id=dataset_id)
+
+
+async def _get_unique_name(session: AsyncSession) -> str:
+    name = f"{choice(ADJECTIVES)}-{choice(BIRD_SPECIES)}"
+    k = len(name)
+    while await session.scalar(
+        select(1).select_from(models.Dataset).where(models.Dataset.name == name)
+    ):
+        name = f"{name[:k]}-{randint(0,10**6-1):06d}"
```

Comment:
This sort of has me wondering what happens if the user has set a random seed in the environment that phoenix is running in. Do we have a way of injecting the random seed here rather than relying on a global random seed?

Reply:
I guess it's okay because `randint` will eventually return a previously unseen value.